### PR TITLE
Update CITATION.cff for latest release

### PR DESCRIPTION
**Pull request by MatthewDaggitt**

### CODE REVIEW

**Comment by MatthewDaggitt** [comment]
Would be good if we could fix #819 as well in this one.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -122,6 +122,11 @@ We recommend you install GHC and Cabal using [GHCup].
 
 If you'd like to use a different version of GHC, you can find the list of versions that we test with in [build-vehicle.yml]. However, be aware that building the Python bindings _requires_ [our preferred version](#the-preferred-version-of-ghc).
 
+Some of the Haskell libraries used by Vehicle also required `gmp` arithmetic library to be installed on the system. If it is not installed on your system you can install it, e.g. on Linux by
+   ```sh
+   sudo apt-get install libgmp3-dev
```
Comment (wenkokke) [inline CONTRIBUTING.md:127]: When are folks encountering this issue? It shouldn't be with a Python install.

Comment (wenkokke) [reply CONTRIBUTING.md:127]: This install command is specific to Ubuntu, which means it's not all that useful. Furthermore, most Haskell installation instructions inside instructions for installing gmp, or do it automatically.

File: CITATION.cff [hunk 2]
```diff
@@ -79,6 +79,18 @@ references:
     year: "2022"
     url: "https://arxiv.org/abs/2202.05207"
 identifiers:
+  - type: url
+    value: "https://github.com/vehicle-lang/vehicle/releases/tag/v0.14.1"
+    description: "The GitHub release URL of tag v0.14.1."
+  - type: url
+    value: "https://gpypi.org/project/vehicle-lang/0.14.1/"
```
Comment (wenkokke) [inline CITATION.cff:86]: This shouldn't say "gpypi".

File: CITATION.cff [hunk 3]
```diff
@@ -79,6 +79,18 @@ references:
...
+    description: "The PyPI release URL of version 0.14.1."
+  - type: url
+    value: "https://github.com/vehicle-lang/vehicle/releases/tag/v0.14.0"
+    description: "The GitHub release URL of tag v0.14.0."
+  - type: url
+    value: "https://gpypi.org/project/vehicle-lang/0.14.0/"
```
Comment (wenkokke) [inline CITATION.cff:92]: This shouldn't say "gpypi".

**Review by wenkokke — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (MatthewDaggitt) [reply CONTRIBUTING.md:127]: > When are folks encountering this issue? It shouldn't be with a Python install.

This was installing it from source on my new work laptop...

**Pull request closed without merging**